### PR TITLE
Add more unit tests for LogEntry fields.

### PR DESCRIPTION
**Pull request by qingling128**
Follow up of https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/pull/273.

### CODE REVIEW

File: test/plugin/test_out_google_cloud_grpc.rb [hunk 1]
```diff
@@ -466,8 +466,78 @@ def get_number(field)
...
+  #         "function" => {
+  #           "stringValue" => "my_function"
+  #         },
+  #         "line" => {
+  #           "numberValue" => 18
+  #         }
+  #       }
+  #     }
+  #   }
+  def assert_json_equal(expected, actual)
```
Comment (jkohen) [inline test/plugin/test_out_google_cloud_grpc.rb]: assert_json_equal_proto highlights the fact that these are different structures.

Comment (qingling128) [reply test/plugin/test_out_google_cloud_grpc.rb]: I thought about this, then realized this method name has to be shared between gRPC and REST. The REST side does not involve proto. I just added some comments for clarification.

Comment (jkohen) [reply test/plugin/test_out_google_cloud_grpc.rb]: How about hash_equal_json or something where the reader understands at the call location, without having to read comments?

Comment (qingling128) [reply test/plugin/test_out_google_cloud_grpc.rb]: Just renamed this to `assert_hash_equal_json`. WDYT?

**Pull request merged**